src/pkg/index.ts

```diff
@@ -0,0 +1,686 @@
+    const response = await fetch(this.getVersionedUrl("txs/signed"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+
+  public async getMessageBoard(data: GetMessageBoardRequest): Promise<MessageResponse> {
+    const response = await fetch(this.getVersionedUrl("messages/getMessageBoard"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+  public async getMessageDirect(data: GetMessageDirectRequest): Promise<MessageResponse> {
+    const response = await fetch(this.getVersionedUrl("messages/getMessageDirect"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+
+  public async getDatapoints(data: GetOracleDataPointRequest): Promise<DatapointResponse> {
+    const response = await fetch(this.getVersionedUrl("oracles/getDatapoints"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+
+  public async getTickers(addresses: string[]): Promise<TickerResponse> {
+    const response = await fetch(this.getVersionedUrl("oracles/getTickers"), this.getRequestBody({ addresses }));
+    return await response.json();
+  }
+
+  public async getCardanoWallet(data: CardanoWalletRequest): Promise<PoolInfo[]> {
+    const response = await fetch(this.getVersionedUrl("pool/cardanoWallet"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+
+  public async multiAssetSupply(assets: Asset[]): Promise<MultiAssetSupplyResponse> {
+    const response = await fetch(this.getVersionedUrl("multiAsset/supply"), this.getRequestBody({ assets }));
+    return await response.json();
+  }
+
+  public async multiAssetMetadata(assets: Asset[]): Promise<MultiAssetMetadataResponse> {
+    const response = await fetch(this.getVersionedUrl("multiAsset/metadata"), this.getRequestBody({ assets }));
+    return await response.json();
+  }
+
+  public async getAssetMintTxs(fingerprint: string): Promise<any> {
+    const response = await fetch(this.getVersionedUrl(`asset/${fingerprint}/mintTxs`));
+    return await response.json();
+  }
+
+  public async validateNFT(fingerprint: string, envName: string): Promise<MultiAssetMetadataResponse> {
+    const response = await fetch(this.getVersionedUrl(`multiAsset/validateNFT/${fingerprint}`), this.getRequestBody({ envName }));
+    return await response.json();
+  }
+
+  public async txStatus(data: TxStatusRequest): Promise<TxStatusResponse> {
+    const response = await fetch(this.getVersionedUrl("tx/status"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+
+  public async policyIds(data: PolicyIDRequest): Promise<PolicyIDResponse> {
+    const response = await fetch(this.getVersionedUrl("multiAsset/policyIdExists"), this.getRequestBody({ data }));
+    return await response.json();
+  }
+
+  public async getStatus(): Promise<StatusResponse> {
+    const response = await fetch(this.getVersionedUrl("status"));
+    return await response.json();
+  }
+
+  public async getFundInfo(): Promise<any> {
```

Comment:
can we add types for this request as well?

Reply:
This one is not suitable to type because the response itself from the service which brings the data is also typed as "any".

Reply:
I see your point, but in this case the response is determined by a file we have on S3, and the format will always be:
```
{
    "currentFund": {
        "id": string,
        "name": string,
        "registrationStart": Date,
        "registrationEnd": Date,
        "votingStart": Date,
        "votingEnd": Date,
        "votingPowerThreshold": string | number
    }
}
```